Aztec/Classes/Libxml2/DOM/TextNode.swift

```diff
@@ -88,25 +94,25 @@ extension Libxml2 {
             }
 
             guard let parent = parent,
-                let nodeIndex = parent.children.indexOf(self) else {
+                let nodeIndex = parent.children.index(of: self) else {
 
                 fatalError("This scenario should not be possible. Review the logic.")
             }
 
-            let preRange = contents.startIndex ..< swiftRange.startIndex
-            let postRange = swiftRange.endIndex ..< contents.endIndex
+            let preRange = contents.startIndex ..< swiftRange.lowerBound
+            let postRange = swiftRange.upperBound ..< contents.endIndex
 
-            if postRange.count > 0 {
-                let newNode = TextNode(text: contents.substringWithRange(postRange))
+            if postRange.lowerBound != postRange.upperBound {
```

Comment:
they change the interface for the range struct so the check needed to be done differently.